ihontaryk/homework_6/unpack.py

```diff
@@ -0,0 +1,85 @@
+def unpack_while_loop(arguments):
+    """
+    Function converts the arguments
+    into a list and  as a result,  it creates a new list from popped elements
+    """
+
+    arg_list = list(arguments)
+    result = list()
+
+    assert arg_list
+
+    while arg_list:
+        element = arg_list.pop()
+        if isinstance(element, (list, tuple)):
+            arg_list.extend(element)
+        elif isinstance(element, dict):
+            arg_list.extend(element.items())
```

Comment:
I think you should not do this, OR extend at once every item to existing collection not to move to queue

```
from itertools import chain
arg_list.extend(chain.from_iterable(element.items()))

```

Reply:
also you have no order guarantee, and if you do that for dict, why you do not do that for set )